lib/rucio/core/rule.py

```diff
@@ -714,7 +715,10 @@ def list_rules(filters={}, *, session: "Session"):
     :raises:        RucioException
     """
 
-    query = session.query(models.ReplicationRule)
+    query = session.query(models.ReplicationRule).join(models.DataIdentifier,
+                                                       (models.ReplicationRule.scope == models.DataIdentifier.scope) &
+                                                       (models.ReplicationRule.name == models.DataIdentifier.name)
+                                                       ).with_entities(models.ReplicationRule, models.DataIdentifier.bytes)
```

Comment:
1. Is there a reason for `with_entities()`? Would it not work if you added `models.DataIdentifier.bytes` as a second argument to `query()`?
 2. Please prefer the `and_()` function to the `&` operator.
 3. Since you’re touching this area, would you like to try to first migrate it to the SQLAlchemy 2.0 syntax (as its own commit, referencing #5247)? I believe that it will prove trivial.

Reply:
1. Initially I was considering using the second argument but it evoked several `TypeError`s in tests. But I'll try to change it with two arguments again
3 .So it should be redone like: 
```
query = select(models.ReplicationRule, models.DataIdentifier.bytes).join(
models.DataIdentifier, and_(
(models.ReplicationRule.scope == models.DataIdentifier.scope),
(models.ReplicationRule.name == models.DataIdentifier.name))
```,
right?

Reply:
Yes, like so. You can remove the parentheses around the comparisons. I hope I’m not misleading you, but a quick attempt of my own appeared to be successful.

Reply:
Please also replace all instances of the `filter()` method with `where()`.

We’ve adopted a different styling convention for SQL queries which improves their readability. Consequently, I would recommend formatting it like so:
```python
stmt = select(
    models.ReplicationRule,
    models.DataIdentifier.bytes
).join(
    models.DataIdentifier,
    and_(
        models.ReplicationRule.scope == models.DataIdentifier.scope,
        models.ReplicationRule.name == models.DataIdentifier.name
    )
)
```